classes/UpgradeSelfCheck.php

```diff
@@ -465,4 +480,64 @@ public function runPrestaShopCoreChecks()
 
         return true;
     }
+
+    /**
+     * Get php min and max compatible versions for target prestashop version
+     *
+     * @return array
+     */
+    public function phpCompatibleVersions()
```

Comment:
This method might be more useful it you could provide it with a parameter Or if it's not this one then maybe you could extract this algo in a simple tool class that could be used somewhere else

Among other things it would allow you to build the range of compatibility that you display in your table (by looping though PHP versions and PS versions you could easily build a matrix of compatibility that you can display in twig, thus it would be much easier to maintain in the future as we would just need to some additional versions in the arrays)